src/main/java/com/cgdpd/library/dto/pagination/PaginationCriteria.java

```diff
@@ -0,0 +1,32 @@
+package com.cgdpd.library.dto.pagination;
+
+import static com.cgdpd.library.util.OptionalUtil.actualOrEmpty;
+import static com.cgdpd.library.validation.Validator.requiredNotNegative;
+import static com.cgdpd.library.validation.Validator.requiredPositive;
+
+import lombok.Builder;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+
+import java.util.Optional;
+
+@Builder
+public record PaginationCriteria(int pageIndex,
+                                 int pageSize,
+                                 Optional<SortParam> sort) {
+
+    public PaginationCriteria(int pageIndex,
+                              int pageSize,
+                              Optional<SortParam> sort) {
+        this.pageIndex = requiredPositive("pageIndex", pageIndex);
```

Comment:
documentation from `PageRequest`

page – zero-based page index, must not be negative.

So this can be zero

```suggestion
        this.pageIndex = requiredNotNegative("pageIndex", pageIndex);
```